### DISCUSSION TITLE
Using R kernel in interactive window, a key shortcut to execute current chunk/cell and line/selection in interactive when language mode in editor is R

### DISCUSSION THREAD

**Discussion by jaakko-paavola** [microsoft/vscode-jupyter#9911] [New Ideas]
I am experimenting with R in VS code using https://github.com/IRkernel/IRkernel and https://github.com/REditorSupport/vscode-R. It seems to work fine otherwise, but a big problem for me is that I cannot keep the language mode in the editor as 'R' and be able to send chunks/cells or current line/selection to the interactive window, because those keyboard shortcuts seem to work only in Python language mode. If I change the language mode to Python, I lose R syntax highlighting, autocompletion and other goodies provided by https://github.com/REditorSupport/vscode-R.

**Comment by DonJayamanne** [comment]
this functionality is currently tracked here https://github.com/microsoft/vscode-jupyter/issues/273
please upvote that issue

**Comment by jaakko-paavola** [comment]
Thanks. Actually I just realized that "Jupyter: Run Selection/Line in Interactive Window" works with R language mode if the When condition has editorLangId == 'r'. Even the cell running commands work if I switch the language mode to Python and then back to R -- until I switch the active tab, after which it stops working.

**Reply by DonJayamanne** [reply]
Thanks, that sounds like a bug in our code. Will look into this.

**Comment by jaakko-paavola** [comment]
Now that the Jupyter extension is officially supporting R (https://github.com/microsoft/vscode-jupyter/blob/113c3e54ac1d3cb81ab6473d1a5fa4a20cce4755/README.md#working-with-other-languages), is there a way to have it send your current R code chunk in an .Rmd file to the interactive window?

I suppose you could edit the setting "jupyter.interactiveWindow.cellMarker.codeRegex": "^(#\\s*%%|#\\s*\\<codecell\\>|#\\s*In\\[\\d*?\\]|#\\s*In\\[ \\])" to also match R markdown cells starting with ```{r} and ending with ```. But when I tried this I did not get it working unless I also changed the setting "jupyter.defaultCellMarker": "# %%" to "```{r}". Also, as the description of the setting says: Cell marker used for delineating a cell in a python file, these cells seem to be only recognized in .py files, not in .Rmd files.
